src/pulsar/mod.rs

```diff
@@ -57,10 +57,21 @@ pub async fn pulsar_cli_run(options: &PulsarCliOpts) -> Result<()> {
             _ => unreachable!(),
         },
         Commands::Monitor(Monitor { all }) => {
-            while let Ok(msg) = engine_api_client.event_monitor().await?.try_next().await {
-                if let Some(event) = msg {
-                    if *all || event.header().threat.is_some() {
-                        logger::terminal::print_event(&event)
+            let mut stream = engine_api_client.event_monitor().await?;
+
+            loop {
+                match stream.next().await {
+                    Some(trytry) => match trytry {
+                        Ok(event) => {
+                            if *all || event.header().threat.is_some() {
+                                logger::terminal::print_event(&event)
+                            }
+                        }
+                        Err(e) => println!("ws error {e}"),
```

Comment:
Should this be `log::error`?

Reply:
changed to return, covering server disconnections